elide-async/src/main/java/com/yahoo/elide/async/models/security/AsyncQueryOperationChecks.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.models.security;
+
+import java.security.Principal;
+import java.util.Optional;
+
+import com.yahoo.elide.annotation.SecurityCheck;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.security.ChangeSpec;
+import com.yahoo.elide.security.RequestScope;
+import com.yahoo.elide.security.checks.OperationCheck;
+
+/**
+ * Operation Checks on the Async Query and Result objects.
+ */
+public class AsyncQueryOperationChecks {
+    @SecurityCheck(AsyncQueryOwner.PRINCIPAL_IS_OWNER)
+    public static class AsyncQueryOwner extends OperationCheck<Object> {
+
+        public static final String PRINCIPAL_IS_OWNER = "Principal is Owner";
+
+        @Override
+        public boolean ok(Object object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
+            Principal principal = ((Principal) requestScope.getUser().getOpaqueUser());
+
+            if(object.getClass().equals(AsyncQuery.class)) {
```

Comment:
Cast to PrincipalOwned.  Then you don't need to handle the two cases differently.  

Also - let's file a ticket to move this to a FilterExpressionCheck

Reply:
I can not remember the reason, but can you explain what will be the advantage of moving this to FilterExpressionCheck vs the OperationCheck?